irr/include/IEventReceiver.h

```diff
@@ -344,6 +344,9 @@ struct SEvent
 
 		//! Type of mouse event
 		EMOUSE_INPUT_EVENT Event;
+
+		//! Is this a simulated mouse event generated by Minetest itself?
+		bool Simulated;
```

Comment:
you could reuse e.g. `ButtonStates` with a flag that indicates simulated or not
but other than that setting this in the constructor sounds like the correct approach

Reply:
> you could reuse e.g. `ButtonStates` with a flag that indicates simulated or not

I fear bugs, consider this:

https://github.com/minetest/minetest/blob/9def45aa8006426a9450203700e7b3a6324dc351/src/gui/guiFormSpecMenu.cpp#L4122-L4126

>  but other than that setting this in the constructor sounds like the correct approach

I now notice that setting it in the constructor probably isn't good either. Consider `CIrrDeviceSDL`, which reuses one `SEvent` variable for all Irrlicht events it creates from SDL events.

https://github.com/minetest/minetest/blob/bc23a610d350dfe81594959f5b238eed4bde0f95/irr/src/CIrrDeviceSDL.cpp#L646-L650

Since union members other than `MouseInput` are used on that variable (at least `KeyInput` & `TouchInput`), `MouseInput.Simulated` might (in theory) already be overwritten when a mouse event is created.

Reply:
> I now notice that setting it in the constructor probably isn't good either. Consider `CIrrDeviceSDL`, which reuses one `SEvent` variable for all Irrlicht events it creates from SDL events.

bad practice IMO, it shouldn't be doing that.
if you want to fix that you could simply insert `irrevent = {}` before it gets initialized.

Reply:
Hmm, shouldn't replacing every occurrence of `SEvent event;` with `SEvent event{};` also work as a solution?